### PR TITLE
Jruby prepare jars

### PR DESCRIPTION
**Pull request by mkristian**
this is basically the fix for #141

### CODE REVIEW

File: jruby-gradle-base-plugin/src/main/groovy/com/github/jrubygradle/JRubyPlugin.groovy [hunk 1]
```diff
@@ -41,30 +40,30 @@ class JRubyPlugin implements Plugin<Project> {
             JRubyExec.updateJRubyDependencies(project)
         }
 
-        project.task('jrubyPrepareGems', type: JRubyPrepareGems) {
+        project.task('jrubyPrepareGems', type: JRubyPrepare) {
```
Comment (ysb33r) [inline jruby-gradle-base-plugin/src/main/groovy/com/github/jrubygradle/JRubyPlugin.groovy]: jrubyPrepareGems just needs to be an empty task with one action added and a dependency added. No need to make it a JRubyPrepare task.

```groovy
Task jrpg = project.task.create 'jrubyPrepareGems'
jrpg.dependsOn 'jrubyPrepare'
jrpg << { logger.info "'jrubyPrepareGems' is deprecated and will be removed in a future version. Use 'jrubyPrepare' instead." }
```

File: jruby-gradle-base-plugin/src/main/groovy/com/github/jrubygradle/JRubyPrepare.groovy [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.file.FileCollection
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+
+/**
+ * @author Schalk W. Cronjé
+ * @author R Tyler Croy
```
Comment (ysb33r) [inline jruby-gradle-base-plugin/src/main/groovy/com/github/jrubygradle/JRubyPrepare.groovy]: Put your own name in here

File: jruby-gradle-base-plugin/src/main/groovy/com/github/jrubygradle/JRubyPrepareGems.groovy [hunk 3]
```diff
@@ -12,44 +13,7 @@ import org.gradle.api.tasks.TaskAction
...
-
-    @TaskAction
-    void copy() {
-        File jrubyJar = JRubyExecUtils.jrubyJar(project.configurations.getByName(JRubyExec.JRUBYEXEC_CONFIG))
-        GemUtils.extractGems(project, jrubyJar, getGems(), outputDir, GemUtils.OverwriteAction.SKIP)
-    }
-
-    private List<Object> gems
+@Deprecated
+class JRubyPrepareGems  extends JRubyPrepare {
```
Comment (ysb33r) [inline jruby-gradle-base-plugin/src/main/groovy/com/github/jrubygradle/JRubyPrepareGems.groovy]: Add c-tor
```groovy
JRubyPrepareGems () {
  super()
  project.logger.info "The 'JRubyPrepareGems' task type is deprecated and will be removed in a future version. Please use 'JRubyPrepare' tasl type instead."
}
```

File: jruby-gradle-base-plugin/src/main/groovy/com/github/jrubygradle/JRubyPrepareJars.groovy [hunk 4]
```diff
@@ -19,7 +19,7 @@ import org.gradle.api.tasks.TaskAction
  * @author R Tyler Croy
  * @author Christian Meier
  */
-@Incubating
+@Deprecated
 class JRubyPrepareJars  extends DefaultTask {
```
Comment (ysb33r) [inline jruby-gradle-base-plugin/src/main/groovy/com/github/jrubygradle/JRubyPrepareJars.groovy]: Same c-tor note as for `JRubyPrepareGems`.

File: jruby-gradle-base-plugin/src/test/groovy/com/github/jrubygradle/JRubyPluginTest.groovy [hunk 5]
```diff
@@ -31,7 +31,7 @@ class JRubyPluginTest {
 
     @Test
```
Comment (ysb33r) [inline jruby-gradle-base-plugin/src/test/groovy/com/github/jrubygradle/JRubyPluginTest.groovy]: Just wondering why we have not yet converted this to SpockFw...

Never mind too much, not needed for this PR.

File: jruby-gradle-base-plugin/src/main/groovy/com/github/jrubygradle/JRubyPlugin.groovy [hunk 6]
```diff
@@ -41,30 +41,28 @@ class JRubyPlugin implements Plugin<Project> {
             JRubyExec.updateJRubyDependencies(project)
         }
 
-        project.task('jrubyPrepareGems', type: JRubyPrepareGems) {
-            group TASK_GROUP_NAME
-            description 'Prepare the gems from the `gem` dependencies, extracts into jruby.installGemDir'
-            gems project.configurations.gems
-            outputDir project.jruby.gemInstallDir
-        }
+        Task jrpg = project.tasks.create 'jrubyPrepareGems'
```
Comment (rtyler) [inline jruby-gradle-base-plugin/src/main/groovy/com/github/jrubygradle/JRubyPlugin.groovy]: This is unnecessary FWIW, you can still use the task creation closure for generic tasks like this, e.g.

```groovy
Task jrpg = project.tasks.create('jrubyPrepareGems') {
  dependsOn 'jrubyPrepare'
  // etc
}
```

This doesn't need to change for this PR, just wanted to bring it up

File: jruby-gradle-base-plugin/src/test/groovy/com/github/jrubygradle/JRubyPluginTest.groovy (hunk 5 above)
Comment (rtyler) [inline jruby-gradle-base-plugin/src/test/groovy/com/github/jrubygradle/JRubyPluginTest.groovy]: #148 for this

**Pull request merged**